### PR TITLE
Fix Issue 18299: std.datetime.date.cmpTimeUnits does not throw a Date…

### PR DESCRIPTION
**Pull request by JackStouffer**
…TimeException

### CODE REVIEW

File: std/datetime/date.d [hunk 1]
```diff
@@ -9986,12 +9986,11 @@ int cmpTimeUnits(string lhs, string rhs) @safe pure
     import std.exception : enforce;
     import std.format : format;
 
-    auto tstrings = timeStrings;
-    immutable indexOfLHS = countUntil(tstrings, lhs);
-    immutable indexOfRHS = countUntil(tstrings, rhs);
+    immutable indexOfLHS = countUntil(timeStrings, lhs);
+    immutable indexOfRHS = countUntil(timeStrings, rhs);
```
Comment (quickfur) [inline std/datetime/date.d]: Just curious, what was the reason for this change?

Comment (JackStouffer) [reply std/datetime/date.d]: No reason to make a second copy of the reference when countUntil takes it by value

Comment (quickfur) [reply std/datetime/date.d]: Fair enough.

**Comment by wilzbach** [comment]
Once https://github.com/dlang/phobos/pull/6070 is merged, CircleCi can be restarted here.

**Pull request merged**